corehq/apps/case_search/xpath_functions/subcase_functions.py

```diff
@@ -0,0 +1,196 @@
+    """True if the initial expression is one of ['<', '<=']"""
+
+    def __post_init__(self):
+        ops = ('>', '=')
+        if self.op not in ops:
+            raise ValueError(f"op must be one of {ops}")
+
+    def as_tuple(self):
+        return (
+            self.index_identifier, serialize(self.subcase_filter), self.op, self.count, self.invert
+        )
+
+    def filter_count(self, count):
+        if self.op == '>':
+            return count > self.count
+        return self.count == count
+
+
+def subcase(domain, node, fuzzy=False):
+    """
+    Supports the following syntax:
+    - subcase-exists('parent', {subcase filter} )
+    - subcase-count('host', {subcase_filter} ) {=, !=, >, <, >=, <=} {integer value}
+    """
+    subcase_query = _parse_normalize_subcase_query(node)
+    ids = _get_parent_case_ids_matching_subcase_query(domain, subcase_query, fuzzy)
+    if subcase_query.invert:
+        if not ids:
+            return filters.match_all()
+        return filters.NOT(filters.doc_id(ids))
+    # uncomment once we are on ES > 2.4
+    # if not ids:
+    #     return filters.match_none()
+    return filters.doc_id(ids)
+
+
+def _get_parent_case_ids_matching_subcase_query(domain, subcase_query, fuzzy=False):
+    """Get a list of case IDs for cases that have a subcase with the given index identifier
+    and matching the subcase predicate filter.
+
+    Only cases with `[>,=] case_count_gt` subcases will be returned.
+    """
+    # TODO: validate that the subcase filter doesn't contain any ancestor filtering
+    from corehq.apps.case_search.filter_dsl import (
+        MAX_RELATED_CASES,
+        build_filter_from_ast,
+    )
+
+    from ..exceptions import TooManyRelatedCasesError
+
+    subcase_filter = build_filter_from_ast(domain, subcase_query.subcase_filter, fuzzy=fuzzy)
+
+    index_query = queries.nested(
+        'indices',
+        queries.filtered(
+            queries.match_all(),
+            filters.AND(
+                filters.term('indices.identifier', subcase_query.index_identifier),
+                filters.NOT(filters.term('indices.referenced_id', ''))  # exclude deleted indices
+            )
+        )
+    )
+    es_query = (
+        CaseSearchES().domain(domain)
+        .filter(index_query)
+        .filter(subcase_filter)
+        .source('indices')
+    )
+
+    if es_query.count() > MAX_RELATED_CASES:
+        raise TooManyRelatedCasesError(
+            _("The related case lookup you are trying to perform would return too many cases"),
+            serialize(subcase_query.subcase_filter)
+        )
+
+    parent_case_id_counter = Counter()
+    for subcase in es_query.run().hits:
+        indices = [index for index in subcase['indices'] if index['identifier'] == subcase_query.index_identifier]
+        if indices:
+            parent_case_id_counter.update([indices[0]['referenced_id']])
+
+    if subcase_query.op == '>' and subcase_query.count <= 0:
+        return list(parent_case_id_counter)
```

Comment:
nitpick: isn't this redundant with the line below?

Reply:
It is but it skips iterating through the data and checking the count for each one.